### ISSUE TITLE
Fixed final time problems don't generate the correct boundary conditions

### ISSUE THREAD

**Issue by msparapa** [Rapid-Design-of-Systems-Laboratory/beluga#148]
Problems with final time fixed don't appear to be generating the correct boundary conditions.

**Comment by msparapa** [comment]
@SeanMatthewNolan I fixed this issue, but I'm not happy with my fix. I was hoping to adjoin the independent variable as a state to all problems regardless of whether or not it was used. I got this too work, but some of my parameters and constraints were redundant leading to a rank deficient Jacobian matrix. Your Armijo line search solved these perfectly fine because it can do the pseudoinverse, but SLSQP for Shooting and also Collocation didn't work because of the noninvertible Jacobian. Until I can figure out what's causing the singularity, I'm not adjoining time as a state.

Also, I implemented a `guess_map_inverse` function to be output by `optimlib` in addition to the `guess_map`. Since `optimlib` is essentially a functor between OCP and BVP problems, we had `guess_map` which maps OCP solutions and guesses to BVP solutions and guesses. We also now have `guess_map_inverse` which takes the BVP solution and maps it back to an OCP solution. Ideally the designer will only see OCP solutions. i.e. `icrm` on brachistochrone will only output the system's states and costates and not the dae states. Those dae states are automatically mapped to the control variables by the `guess_map_inverse`. Ideally all outputs by beluga should be identical regardless of whether or not ICRM, traditional, or direct methods are used.
